Automattic/wp-calypso
# Components: Remove loading.gif in favor of Spinner component

This pull request seeks to remove all remaining references of `loading.gif` from Calypso:

![loading.gif](https://wordpress.com/calypso/images/loading.gif)

There were only two remaining usages, which have been updated to use the [`<Spinner /> component`](https://github.com/Automattic/wp-calypso/tree/master/client/components/spinner) instead.

Changes also include some refactoring to `<Spinner />`, particularly to ensure that it's rendered as a pure component, which may improve performance slightly (particularly for media uploading where many spinners may exist simultaneously).

__Testing instructions:__

Verify that there are no regressions in display of loading indicator for affected stats placeholders and Customize loading indicator.

1. Navigate to [Stats insights](http://calypso.localhost:3000/stats/insights)
2. Select a site, if prompted
3. Note that under "Latest Post Summary", while post is loading, a centered spinner is shown
4. Click Customize in the sidebar
5. Note that the "Loading customizer" text includes a centered spinner above

/cc @timmyc re: removed `is-spinner` class selector in stats styles (couldn't find any references to justify distinction between `is-spinner` and `is-void`).

Test live: https://calypso.live/?branch=remove/spinner-gif
